transcom/mymove
# INT B-20450 boat customer shipment card

## [B-20450](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20450)

## Summary

This is to add/modify boat shipment card to the customer's app and make some adjustments to the boat shipment form flow.

- Added year validation (min 1700 to max current year +2).
- When editing an existing boat shipment from meeting dimension requirements to not meeting the requirements -> it will display a modal that will delete the current boat shipment and redirect user to a HHG form with existing user inputs if the boat shipment was completed before.
- Added more automated tests.
- Added a check in the backend for shipmentNum/seq num to catch an edge case.
- Added boat shipment card in review shipments page.
- Updated boat shipment UI for shipment lists.

### How to test

1. As a customer -> create a move with a boat shipment(put 22 feet for at least one to make it a boat shipment).
2. Verify you can see the information you have entered in the boat shipment card(see image).
3. Verify the boat shipment row matches other shipments(see image).
4. Click edit for boat shipment and change to have dimensions be 2 feet.
5. You should see a modal to delete shipment and continue.
6. Click "Delete & Continue" and fill out the form.
7. Verify that the boat shipment has been deleted and you have created a HHG shipment.

## Screenshots
![Screenshot 1](https://github.com/user-attachments/assets/888c38c2-5922-4123-84a6-d2dd54d9b24e)
![Screenshot 2](https://github.com/user-attachments/assets/a1880257-f919-48c7-bd51-40b6fa2fe011)
![Screenshot 3](https://github.com/user-attachments/assets/a21dfa0b-0249-4c95-bb82-8b2faac5ee12)
